homeassistant/components/here_travel_time/config_flow.py

```diff
@@ -0,0 +1,360 @@
+    TRAVEL_MODE_PUBLIC_TIME_TABLE,
+    TRAVEL_MODES,
+    UNITS,
+)
+from .sensor import (
+    CONF_DESTINATION_ENTITY_ID,
+    CONF_DESTINATION_LATITUDE,
+    CONF_DESTINATION_LONGITUDE,
+    CONF_ORIGIN_ENTITY_ID,
+    CONF_ORIGIN_LATITUDE,
+    CONF_ORIGIN_LONGITUDE,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def is_dupe_import(
+    entry: config_entries.ConfigEntry,
+    user_input: dict[str, Any],
+    options: dict[str, Any],
+) -> bool:
+    """Return whether imported config already exists."""
+    # Check the main data keys
+    if any(
+        user_input[key] != entry.data[key]
+        for key in (CONF_API_KEY, CONF_MODE, CONF_NAME)
+    ):
+        return False
+
+    # Check origin
+    for key in (
+        CONF_DESTINATION,
+        CONF_ORIGIN,
+        CONF_DESTINATION_ENTITY_ID,
+        CONF_ORIGIN_ENTITY_ID,
+    ):
+        if user_input.get(key) != entry.data.get(key):
+            return False
+
+    # We have to check for options that don't have defaults
+    for key in (
+        CONF_TRAFFIC_MODE,
+        CONF_UNIT_SYSTEM,
+        CONF_ROUTE_MODE,
+        CONF_TIME_TYPE,
+        CONF_TIME,
+    ):
+        if options.get(key) != entry.options.get(key):
+            return False
+
+    return True
+
+
+def validate_api_key(api_key: str) -> None:
+    """Validate the user input allows us to connect."""
+    known_working_origin = [38.9, -77.04833]
+    known_working_destination = [39.0, -77.1]
+    RoutingApi(api_key).public_transport_timetable(
+        known_working_origin,
+        known_working_destination,
+        True,
+        [
+            RouteMode[ROUTE_MODE_FASTEST],
+            RouteMode[TRAVEL_MODE_CAR],
+            RouteMode[TRAFFIC_MODE_ENABLED],
+        ],
+        arrival=None,
+        departure="now",
+    )
+
+
+def get_user_step_schema(data: dict[str, Any]) -> vol.Schema:
+    """Get a populated schema or default."""
+    return vol.Schema(
+        {
+            vol.Optional(CONF_NAME, default=data.get(CONF_NAME, DEFAULT_NAME)): cv.string,
+            vol.Required(CONF_API_KEY, default=data.get(CONF_API_KEY)): cv.string,
+            vol.Optional(CONF_MODE, default=data.get(CONF_MODE, TRAVEL_MODE_CAR)): vol.In(TRAVEL_MODES),
+        }
+    )
+
+
+class HERETravelTimeConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for HERE Travel Time."""
+
+    VERSION = 1
+
+    _config: dict[str, Any] | None
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> HERETravelTimeOptionsFlow:
+        """Get the options flow."""
+        return HERETravelTimeOptionsFlow(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors = {}
+        user_input = user_input or {}
+        if user_input:
+            if self.source == config_entries.SOURCE_IMPORT:
```

Comment:
My impression is that this check should not be needed and that what you be missing is that `async_import` should be renamed `async_step_import` so that it is invoked first.  I think the `sun` component may be a good one to model after. This is also related to my other comment about import moving to init.

Reply:
Done